backend/libraries/types/can.did

```diff
@@ -528,6 +528,7 @@ type Message = record {
     content : MessageContent;
     replies_to : opt ReplyContext;
     reactions : vec record { text; vec UserId };
+    tips : vec record { CanisterId; vec record { UserId; nat } };
```

Comment:
@julianjelfs here

Reply:
The canisterId is the ledger canister, then for each ledger there is a list of UserId + amount tipped pairs.

Reply:
OK so that's like a list of tuples? So if the same user tips twice we get two (and we can either choose to aggregate those or not on the front end)?

Reply:
Its a list of tuples but it already aggregates on the backend, so you only see the number of unique users.